src/swell/tasks/ingest_background.py

```
@@ -0,0 +1,169 @@
+            
+            if not os.path.exists(config_path):
+                self.logger.error(f"Config file not found for {bg_name} at {config_path}")
+                total_failed += 1
+                continue
+                
+            # Load the YAML config
+            with open(config_path, 'r') as f:
+                bg_config = yaml.safe_load(f)
+                
+            # Ingestion
+            ingested, failed = self.process_background_config(bg_config, bg_name, window_begin, dry_run)
+            
+            total_ingested += len(ingested)
+            total_failed += len(failed)
+
+        self.logger.info("="*60)
+        self.logger.info("INGESTION SUMMARY")
+        self.logger.info("="*60)
+        if dry_run:
+            self.logger.info(f"Would ingest: {total_ingested} files")
+            self.logger.info(f"Would fail: {total_failed} files")
+        else:
+            self.logger.info(f"Successfully ingested: {total_ingested} files")
+            self.logger.info(f"Failed: {total_failed} files")
+
+    def process_background_config(self, config, bg_name, date, dry_run):
+        """Process a single background configuration file."""
+        ingested = []
+        failed = []
+        
+        # Extract metadata
+        bg_metadata = config.get('bg_to_ingest', {})
+        model = bg_metadata.get('model')
+        experiment = bg_metadata.get('experiment')
+        resolution = bg_metadata.get('resolution')
+        file_type = bg_metadata.get('file_type')
+        step = bg_metadata.get('step', 'P0H')  # Default to analysis time
+        
+        retrieval_method = config.get('retrieval_method')
+        source_pattern = config.get(f'{retrieval_method}_source')
+        
+        if not source_pattern:
+            self.logger.error(f"No source pattern found for method '{retrieval_method}' in {bg_name}.yaml")
+            return ingested, [(bg_name, "Missing source pattern")]
```

Should this be a task-ending error in most cases? If so, we should raise an informative exception here. If not -- i.e., if it's OK that source patterns are missing sometimes --- this should probably be changed to a `logger.warn`.

---

Correct, fixed it.